Xwt.XamMac/Xwt.Mac/WindowFrameBackend.cs

```diff
@@ -164,32 +164,37 @@ public void Present ()
 		}
 
 		WindowFrameEvent eventsEnabled;
+		NSObject didResizeObserver, didMoveObserver, willCloseObserver;
 
 		void IBackend.EnableEvent (object eventId)
 		{
 			if (eventId is WindowFrameEvent) {
 				var @event = (WindowFrameEvent)eventId;
 				switch (@event) {
 				case WindowFrameEvent.BoundsChanged:
-					Window.DidResize += HandleDidResize;
-					Window.DidMove += HandleDidResize;
+					didResizeObserver = NSNotificationCenter.DefaultCenter.AddObserver (NSWindow.ResizedNotification, HandleDidResize, Window);
+					didMoveObserver = NSNotificationCenter.DefaultCenter.AddObserver (NSWindow.MovedNotification, HandleDidResize, Window);
 					break;
 				case WindowFrameEvent.Hidden:
 					EnableVisibilityEvent (@event);
-					Window.WillClose += OnWillClose;
+					willCloseObserver = NSNotificationCenter.DefaultCenter.AddObserver (NSWindow.WillCloseNotification, OnWillClose, Window);
 					break;
 				case WindowFrameEvent.Shown:
 					EnableVisibilityEvent (@event);
 					break;
 				case WindowFrameEvent.CloseRequested:
-					Window.WindowShouldClose = OnShouldClose;
+					// NOTE: this works only if the wrapped window has no delegate
+					if (Window.Delegate == null && Window.WeakDelegate == null) // don't replace existing delegates
+						Window.WindowShouldClose = OnShouldClose;
+					else
+						return; // skip eventEnabled update to avoid touching WindowShouldClose when disabling events
```

Comment:
Should we explicitly throw here?

Reply:
I'd say no, this might break client logic and catching when subscribing to events is really ugly (but would be required then). I think it's ok if the event just wont ever fire. It's a very specific scenario.

Reply:
oh, but this needs another fix...